packages/contracts/contracts/test/Colosseum.t.sol

```diff
@@ -1553,3 +1553,179 @@ contract Colosseum_ValidatorSystemUpgrade_Test is Colosseum_Initializer {
+        if (sender == trusted) {
+            if (blockNumber < targetBlockNumber - 1) {
+                return keccak256(abi.encode(blockNumber));
+            } else {
+                return keccak256(abi.encode());
+            }
+        }
+
+        // If challenger, correct output always
+        if (blockNumber == targetBlockNumber) {
+            return ColosseumTestData.TARGET_OUTPUT_ROOT;
+        } else {
+            return keccak256(abi.encode(blockNumber));
+        }
+    }
+
+    function _newSegments(
+        address sender,
+        uint8 turn,
+        uint256 segStart,
+        uint256 segSize
+    ) private view returns (bytes32[] memory) {
+        uint256 segLen = colosseum.segmentsLengths(turn - 1);
+
+        bytes32[] memory arr = new bytes32[](segLen);
+
+        for (uint256 i = 0; i < segLen; i++) {
+            uint256 n = segStart + i * (segSize / (segLen - 1));
+            arr[i] = _getOutputRoot(sender, n);
+        }
+
+        return arr;
+    }
+
+    function _getFirstSegments() private view returns (bytes32[] memory) {
+        Types.CheckpointOutput memory targetOutput = oracle.getL2Output(targetOutputIndex);
+        uint256 end = targetOutput.l2BlockNumber;
+        uint256 start = end - oracle.SUBMISSION_INTERVAL();
+
+        bytes32[] memory segments = _newSegments(challenger, 1, start, end - start);
+
+        return segments;
+    }
+
+    function test_createChallenge_succeeds() public {
+        bytes32[] memory segments = _getFirstSegments();
+        vm.prank(challenger);
+        colosseum.createChallenge(targetOutputIndex, bytes32(0), 0, segments);
+    }
+
+    function test_createChallenge_mptFirstOutputIndex_reverts() public {
+        bytes32[] memory segments = _getFirstSegments();
+
+        // upgrade validatorManager with targetOutputIndex
+        address valMgrAddress = address(valMgr);
+        constructorParams._mptFirstOutputIndex = targetOutputIndex;
+        MockValidatorManager mockValMgrImpl = new MockValidatorManager(constructorParams);
+        vm.prank(multisig);
+        Proxy(payable(valMgrAddress)).upgradeTo(address(mockValMgrImpl));
+
+        vm.prank(challenger);
+        vm.expectRevert(IValidatorManager.MptFirstOutputRestricted.selector);
+        colosseum.createChallenge(targetOutputIndex, bytes32(0), 0, segments);
+    }
+
+    function test_createChallenge_upgradeMptFirstOutputIndex_succeeds() public {
+        bytes32[] memory segments = _getFirstSegments();
+
+        // upgrade validatorManager with targetOutputIndex + 1
+        address valMgrAddress = address(valMgr);
+        constructorParams._mptFirstOutputIndex = targetOutputIndex + 1;
+        MockValidatorManager mockValMgrImpl = new MockValidatorManager(constructorParams);
+        vm.prank(multisig);
+        Proxy(payable(valMgrAddress)).upgradeTo(address(mockValMgrImpl));
+
+        vm.prank(challenger);
+        colosseum.createChallenge(targetOutputIndex, bytes32(0), 0, segments);
```

Comment:
How about adding an assertEq statement that checks `MPT_FIRST_OUTPUT_INDEX` and `targetOutputIndex + 1` are same?